pkg/clientwrappers/dataflow/dataflow.go

```diff
@@ -293,12 +293,8 @@ func SyncDataflow(
 
 	processGroups = append(processGroups, *pGEntity)
 	if isParameterContextChanged(parameterContext, processGroups) {
-		// unschedule processors
-		_, err := nClient.UpdateFlowProcessGroup(nigoapi.ScheduleComponentsEntity{
-			Id:    flow.Status.ProcessGroupID,
-			State: "STOPPED",
-		})
-		if err := clientwrappers.ErrorUpdateOperation(log, err, "Stop flow"); err != nil {
+		// unschedule dataflow
+		if err := UnscheduleDataflow(flow, config); err != nil {
```

Comment:
Can you add a log message here indicating that nifikop is stopping the entire flow to update the param context? This seems like a good thing to capture in the logs. :)

Reply:
I added them in several functions. Is it good?